get_security_annotations.py

```diff
@@ -176,50 +122,140 @@ def log_event_attributes(alert, threat_intel_record, asset, threat_detection, ev
     print(json.dumps(row))
 
 
-def process_events(alert, threat_detection, threat_intel_record, asset, events_iterator, previous_max_event_modified_at, max_event_modified_at):
+def main():
+    # read stored cursor of last security event item from previous script run (if any)
+    # to process only new or modified events in current script run
+    previous_events_end_cursor = read_events_end_cursor()
+
+    # USE BULK LOADING RESOURCES ON API TO PRELOAD EVENTS AND IMPORTANT REFERRED OBJECTS
+
+    # remember events
+    events = []
+    # cached contextual objects - alerts, threat detections, threat intel records, assets
+    cached_context_objects = {}
+
+    threat_detection_ids = []
+    affected_asset_ids = []
+
+    # iterate over all security events sorted by unique modification sequence number ensuring no event is missed
+    events_iterator = CollectionIterator(
+        collection_url_path=build_event_with_threat_detections_ids_url(),
+        authorization_fn=authorization,
+        query_params={
+            "sort": "modificationSequenceNumber",
+        },
+        cursor=previous_events_end_cursor
+    )
+    # remember events and get ids of related object (threat detections and assets) to "bulk load" them
     while events_iterator.has_next():
         event = events_iterator.next()
```

Comment:
Once the iterator interface is implemented  the loop can be simplified as follows:

```suggestion
    for event in events_iterator:
```